### PR TITLE
Fix example code, enable mid-circuit measurements.

### PR DESCRIPTION
**Pull request by smite**
* Routing method now returns a tuple, example code was not updated to reflect this. Now fixed.
* Remove validation check and routing quirks that prevent mid-circuit measurements.
* Update docs.

COMP-1490

### CODE REVIEW

File: docs/user_guide.rst [hunk 1]
```diff
@@ -277,22 +292,42 @@ The below table summarises the currently available options:
...
+     - :class:`float` | :class:`None`
      - 1.0
      - Set server-side circuit disqualification threshold. If any circuit in a job is estimated to take longer than the
        shortest T2 time of any qubit used in the circuit multiplied by this value, the server will reject the job.
        Setting this value to ``0.0`` will disable circuit duration check.
        The default value ``None`` means the server default value will be used.
-   * - `heralding_mode`
-     - :py:class:`~iqm_client.models.HeraldingMode`
+   * - :attr:`heralding_mode`
+     - :class:`~iqm.iqm_client.models.HeraldingMode`
```
Comment (freetonik) [inline docs/user_guide.rst:325]: I was wondering why we used `ZEROS` and not `ZEROES` since most of our docs tend to use British English :)

**Review by freetonik — APPROVED** [review]

File: docs/user_guide.rst (hunk 1 above)
Comment (smite) [reply docs/user_guide.rst:325]: We haven't standardized the spelling everywhere I guess.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,12 +2,18 @@
 Changelog
 =========
 
+Version 14.4
+============
+
+* Broken example code fixed. `#135 <https://github.com/iqm-finland/cirq-on-iqm/pull/135>`_
+* Allow mid-circuit measurements. `#135 <https://github.com/iqm-finland/cirq-on-iqm/pull/135>`_
```
Comment (freetonik) [inline CHANGELOG.rst:9]:
```suggestion
* Allow mid-circuit measurements. `#135 <https://github.com/iqm-finland/cirq-on-iqm/pull/135>`_
* Broken example code fixed. `#135 <https://github.com/iqm-finland/cirq-on-iqm/pull/135>`_
```

My high school English teacher always said to start with the stronger point :D

**Review by freetonik — APPROVED** [review]

File: CHANGELOG.rst (hunk 2 above)
Comment (smite) [reply CHANGELOG.rst]: Fixed.

**Pull request merged**